hail/src/main/scala/is/hail/expr/ir/LowerMatrixIR.scala

```diff
@@ -312,6 +312,51 @@ object LowerMatrixIR {
     case MatrixExplodeRows(child, path) => TableExplode(lower(child), path)
 
     case mr: MatrixRead => mr.lower()
+
+    case MatrixAggregateColsByKey(child, entryExpr, colExpr) =>
+      val colKey = child.typ.colKey
+
+      val idx1 = Symbol(genUID())
+      val idx2 = Symbol(genUID())
+      val idx3 = Symbol(genUID())
+      val idx4 = Symbol(genUID())
```

Comment:
Are there more informative names you could give for these index variables?

Reply:
yes, sure